src/Model/Wijkanders.hs

```diff
@@ -48,13 +50,16 @@ import           Util                                     ( menusToEitherNoLunch
                                                           , removeWhitespaceTags
                                                           )
 
-hasDate :: Day -> ByteString -> Bool
-hasDate d =
-  let d' = view gregorian d
-  in  isJust . maybeResult . parse
-        (skipMany (skip (not . W8.isDigit)) *> string
-          (fromString (show (ymdDay d') <> "/" <> show (ymdMonth d')))
-        )
+-- | Looks for strings looking like dates, dd/mm where d and m are digits.
+-- ..and gives them in another order to play nice with the
+-- YearMonthDay constructor.
+hasDate :: ByteString -> Maybe (Months, Days)
+hasDate = maybeResult . parse
+  (   ((.) . (.)) swap (,)
+  <$> (skipMany (skip (not . W8.isDigit)) *> integerParser)
+  <*> (string (B8.pack "/") *> integerParser)
+  )
```

Comment:
if this is some kind of computation that is a monad: use do notation and get rid of all the point free stuff :)

Reply:
It isn't a computation imo, but on the other hand, it is quote convoluted and it's more about my fascination for pointless style than actually creating readable code. :) So I'll probably rewrite it in a more readable fashion in a few days.

Reply:
I find writing parsers (using parser combinators) that are readable quite hard.

Reply:
So here's another version of the same code: https://github.com/dtekcth/mat-chalmers/pull/89/commits/dc42e1d4a69fb090fbdd44bbf27a11ec0498090f#diff-9167874cd996e771711a5e4343d80061R56